unit-tests/src/org/commcare/android/database/SqlFileBackedStorageTests.java

```diff
@@ -0,0 +1,245 @@
+            } else if (count == 1) {
+                idOne = i.nextID();
+            } else if (count == 2) {
+                removeTwoEntries(idOne, i.nextID(), userFixtureStorage);
+            } else {
+                // seems to be required; otherwise iterator loops forever.
+                // not sure if it is a robolectric bug or a bug in our iterator that comes up when we iterate and delete at the same time
+                break;
+            }
+
+            count++;
+        }
+    }
+
+    private void removeOneEntry(int id, IStorageUtility<FormInstance> userFixtureStorage) {
+        String fixtureFilename =
+                ((SqlFileBackedStorage<FormInstance>)userFixtureStorage).getEntryFilenameForTesting(id);
+        File fixtureFile = new File(fixtureFilename);
+        Assert.assertTrue(fixtureFile.exists());
+        userFixtureStorage.remove(id);
+        Assert.assertTrue(!fixtureFile.exists());
+    }
+
+    private void removeTwoEntries(int idOne, int idTwo, IStorageUtility<FormInstance> userFixtureStorage) {
+        ArrayList<Integer> toRemoveList = new ArrayList<>();
+        toRemoveList.add(idOne);
+        toRemoveList.add(idTwo);
+
+        String fixtureOneFilename =
+                ((SqlFileBackedStorage<FormInstance>)userFixtureStorage).getEntryFilenameForTesting(idOne);
+        String fixtureTwoFilename =
+                ((SqlFileBackedStorage<FormInstance>)userFixtureStorage).getEntryFilenameForTesting(idTwo);
+        File fixtureFileOne = new File(fixtureOneFilename);
+        File fixtureFileTwo = new File(fixtureTwoFilename);
+
+        Assert.assertTrue(fixtureFileOne.exists());
+        Assert.assertTrue(fixtureFileTwo.exists());
+
+        ((SqlFileBackedStorage<FormInstance>)userFixtureStorage).remove(toRemoveList);
+
+        Assert.assertTrue(!fixtureFileOne.exists());
+        Assert.assertTrue(!fixtureFileTwo.exists());
+    }
+
+    @Test
+    public void testAdd() {
+        IStorageUtilityIndexed<FormInstance> userFixtureStorage = sandbox.getUserFixtureStorage();
+
+        int entryCount = 0;
+        for (IStorageIterator i = userFixtureStorage.iterate(); i.hasMore(); ) {
+            int id = i.nextID();
+            FormInstance fixture = (FormInstance)i.nextRecord();
+            System.out.println(fixture.getName());
+            entryCount++;
+        }
+        System.out.println(entryCount + "");
+    }
+
+    @Test
+    public void testRecordLookup() {
+        SqlFileBackedStorage<FormInstance> userFixtureStorage =
+                CommCareApplication._().getCurrentApp().getFileBackedStorage("fixture", FormInstance.class);
+
+        // TODO PLM: this should return a result. I think the ':' isn't escaped properly in the query
+        Vector<FormInstance> shouldHaveFormResults = userFixtureStorage.getRecordsForValues(new String[]{FormInstance.META_ID}, new String[]{"commtrack:programs"});
+        Assert.assertTrue(shouldHaveFormResults.size() == 0);
```

Comment:
Any thoughts on this ^?
A fixture with an instance id of  `commtrack:programs` should exist. I think the `:` is messing up the query. We should really look into this!